pkg/gateway/lb/handler/grpc/handler.go

```diff
@@ -153,6 +153,14 @@ func (s *server) SearchByID(ctx context.Context, req *payload.Search_IDRequest)
 		}
 		return nil, status.WrapWithNotFound(fmt.Sprintf("SearchByID API uuid %s's object not found", req.GetId()), err, info.Get())
 	}
+	vl := len(vec.GetVector())
+	if vl < 2 {
```

Comment:
`2` is defined as variable in internal/core/ngt/option.go (https://github.com/vdaas/vald/blob/e74596bd8953ac58384e9f98ac245261bc935c72/internal/core/ngt/option.go#L55) but here, it is hard coded. is it okay?

Reply:
I think it's okay for now, internal/core/ngt/minimumDimensionSize is a core/ngt specific option, the gateway doesn't depend on them, if we add a new algorithm, we should add a global standard constraint to it.

Reply:
i see. it's your choice but i think it is better to give a meaningful name for it.

Reply:
I fixed your suggestion, I created internal/core/algorithm package and move ngt package into it